decentraland/unity-explorer
# Textures leaked when leaving and entering Genesis plaza (#2129)

https://github.com/decentraland/unity-explorer/issues/2129

## What does this PR change?

This one is occurring when it tries to download the texture from a content server and the URL it gets before downloading is different from the URL it used when downloaded that same texture from another content server. In the cache, textures are compared using the URL so the second time the texture is required, in case the URL to use is different from the one used before, it will not find the cached instance and will download another one. Potentially, there can be as many cached texture instances as content servers are available.
The fix consists in using the file hash instead of the URL when comparing textures in the chache.

## How to test the changes?

Using Unity editor and the Memory profiler.

1. Launch the explorer
2. Capture memory with the Memory profiler.
3. Teleport from Genesis to Olavra (or any other small world).
4. Teleport to Genesis.
5. Teleport to the previous world.
6. Repeat several times.
7. Capture memory with the Memory profiler.

Compare both captures.
Look for the texture with orange buttons (bafkreigtdgo3eqskps45xwkl4zby4fldpdwt6ksybbjhueowxczdsqp42y).

It should not be there, no new instances of the instance were created.
Close the first capture, check that the instance is in the capture.
